### PR TITLE
Handle db-doc deletions

### PR DESCRIPTION
**Pull request by dianabarsan**
# Description

Adds support to get subject ids from tombstones when assessing db-doc authorization.

medic/cht-core#6164

# Code review items

- Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- Documented: Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by dianabarsan** [comment]
Testing performance with two users against a local deployment:

| User   | Load | Proposed (avg) | Master (avg) | 3.7 (avg) |
|--------|------|----------------|--------------|----------|
| Medium | Idle |18ms| 14ms           | 39ms         |
| Medium | Load | 80ms | 60ms           | 135ms        |
| Large  | Idle |18ms| 14ms           | 260ms        |
| Large  | Load | 80ms |60ms           | 820ms        |

User *medium* - has 1000 contacts, 9000 docs total
User *large* - has 8000 contacts, 18000 docs total

Idle = running a custom script against an idle database that requests docs the users can see. 
Load = both users replicating at the same time.

**Comment by dianabarsan** [comment]
Hey @garethbowen . Would you mind having a look, please? Thank you!

File: api/src/services/authorization.js [hunk 1]
```diff
@@ -217,26 +233,34 @@ const findContactsByReplicationKeys = (replicationKeys) => {
...
   return db.medic
     .query('medic-client/contacts_by_reference', { keys })
     .then(result => {
-      const docIds = replicationKeys.map(replicationKey => {
-        const found = result.rows.find(row => row.key[1] === replicationKey);
-        return found && found.id || replicationKey;
+      let docIds = [];
+      replicationKeys.forEach(replicationKey => {
+        const found = result.rows.filter(row => row.key[1] === replicationKey);
+        docIds.push(...found.length ? found.map(row => row.id) : [replicationKey]);
```
Comment (garethbowen) [inline api/src/services/authorization.js]: I find this section confusing. I think it would help if the `map` was chained after the `filter` rather than buried inside a ternary. Also the spread operator on the ternary result is hard to parse. And there's no point creating an array for `replicationKey` then spreading it again.

I know it's more lines but what about something like...

```js
const keys = result.rows.filter(row => row.key[1] === replicationKey).map(row => row.id);
if (keys.length) {
  docIds.push(...keys);
} else {
  docIds.push(replicationKey);
}
```

Comment (dianabarsan) [reply api/src/services/authorization.js]: Yea, agree. I'll change it tomorrow. I originally wrote it similarly. I secretly love one-liners.
